clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1272,6 +1272,11 @@ private CompletedFetch initializeCompletedFetch(CompletedFetch nextCompletedFetc
                 log.debug("Received unknown leader epoch error in fetch for partition {}", tp);
             } else if (error == Errors.UNKNOWN_SERVER_ERROR) {
                 log.warn("Unknown error fetching data for topic-partition {}", tp);
+            } else if (error == Errors.CORRUPT_MESSAGE) {
+                throw new KafkaException("Encountered corrupt message when fetching offset "
```

Comment:
While we're at it, could we add the fetch offset to the IllegalStateException message below and the `UNKNOWN_SERVER_ERROR` log message above?

Reply:
Done.